### ISSUE TITLE
After a reconnection interfaces don't become `active` again

### ISSUE THREAD

**Issue by viniarck** [kytos-ng/topology#34]
I've stumbled recently upon this problem when I was exploring some interface status events. It turns out that after a switch reconnect the interfaces will have their status `active: False`, it was expected that after a reconnection and based on port description and port status events the status should be sent to this NApp ` topology` accordingly. 

### How to reproduce

1 - start kytos `kytosd -f  -E`
2 - start mininet `sudo mn -c && sudo mn --controller=remote,ip=127.0.0.1,port=6653 --switch=ovsk,protocols=OpenFlow13 --topo=linear,3`
3 - wait for all switches to complete the handshake, double check the topology `/api/kytos/topology/v3` endpoint confirming interfaces and links are active. 
4 - restart mininet to force switches to reconnect
5 - notice that interfaces won't be `active` like illustrated in the diff below:

```
--- before.json 2021-12-21 14:47:53.193794855 -0300
+++ after.json  2021-12-21 14:49:28.463802775 -0300
@@ -2,10 +2,10 @@
     "topology": {
         "links": {
             "4d42dc0852278accac7d9df15418f6d921db160b13d674029a87cef1b5f67f30": {
-                "active": true,
+                "active": false,
                 "enabled": true,
                 "endpoint_a": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:02:3",
                     "link": "4d42dc0852278accac7d9df15418f6d921db160b13d674029a87cef1b5f67f30",
@@ -35,7 +35,7 @@
                     "uni": false
                 },
                 "endpoint_b": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:03:2",
                     "link": "4d42dc0852278accac7d9df15418f6d921db160b13d674029a87cef1b5f67f30",
@@ -68,10 +68,10 @@
                 "metadata": {}
             },
             "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260": {
-                "active": true,
+                "active": false,
                 "enabled": true,
                 "endpoint_a": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:02:2",
                     "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
@@ -101,7 +101,7 @@
                     "uni": false
                 },
                 "endpoint_b": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:01:2",
                     "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
@@ -137,7 +137,7 @@
         "switches": {
             "00:00:00:00:00:00:00:01": {
                 "active": true,
-                "connection": "127.0.0.1:39186",
+                "connection": "127.0.0.1:39194",
                 "data_path": "s1",
                 "dpid": "00:00:00:00:00:00:00:01",
                 "enabled": true,
@@ -150,7 +150,7 @@
                         "id": "00:00:00:00:00:00:00:01:1",
                         "link": "",
                         "lldp": true,
-                        "mac": "b6:35:54:58:e1:0f",
+                        "mac": "ae:a8:ec:3b:d9:4f",
                         "metadata": {},
                         "name": "s1-eth1",
                         "nni": false,
@@ -158,17 +158,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 516,
+                            "rx_bytes": 90,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 6,
-                            "tx_bytes": 2112,
+                            "rx_packets": 1,
+                            "tx_bytes": 266,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 3
                         },
                         "switch": "00:00:00:00:00:00:00:01",
                         "type": "interface",
@@ -180,7 +180,7 @@
                         "id": "00:00:00:00:00:00:00:01:2",
                         "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
                         "lldp": true,
-                        "mac": "46:18:91:57:72:d1",
+                        "mac": "6a:c4:93:f2:67:8a",
                         "metadata": {},
                         "name": "s1-eth2",
                         "nni": true,
@@ -188,17 +188,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 2112,
+                            "rx_bytes": 266,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 15,
-                            "tx_bytes": 2112,
+                            "rx_packets": 3,
+                            "tx_bytes": 266,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 3
                         },
                         "switch": "00:00:00:00:00:00:00:01",
                         "type": "interface",
@@ -210,7 +210,7 @@
                         "id": "00:00:00:00:00:00:00:01:4294967294",
                         "link": "",
                         "lldp": true,
-                        "mac": "3e:1e:ec:77:43:43",
+                        "mac": "06:8d:c5:58:73:42",
                         "metadata": {},
                         "name": "s1",
                         "nni": false,
@@ -248,7 +248,7 @@
             },
             "00:00:00:00:00:00:00:02": {
                 "active": true,
-                "connection": "127.0.0.1:39190",
+                "connection": "127.0.0.1:39200",
                 "data_path": "s2",
                 "dpid": "00:00:00:00:00:00:00:02",
                 "enabled": true,
@@ -261,7 +261,7 @@
                         "id": "00:00:00:00:00:00:00:02:1",
                         "link": "",
                         "lldp": true,
-                        "mac": "86:17:af:21:31:07",
+                        "mac": "5a:12:e4:a1:f3:5b",
                         "metadata": {},
                         "name": "s2-eth1",
                         "nni": false,
@@ -269,17 +269,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 516,
+                            "rx_bytes": 176,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 6,
-                            "tx_bytes": 2112,
+                            "rx_packets": 2,
+                            "tx_bytes": 266,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 3
                         },
                         "switch": "00:00:00:00:00:00:00:02",
                         "type": "interface",
@@ -291,7 +291,7 @@
                         "id": "00:00:00:00:00:00:00:02:2",
                         "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
                         "lldp": true,
-                        "mac": "9a:c2:af:3b:9e:c3",
+                        "mac": "f6:78:2b:90:cf:61",
                         "metadata": {},
                         "name": "s2-eth2",
                         "nni": true,
@@ -299,17 +299,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 2112,
+                            "rx_bytes": 180,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 15,
-                            "tx_bytes": 2112,
+                            "rx_packets": 2,
+                            "tx_bytes": 266,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 3
                         },
                         "switch": "00:00:00:00:00:00:00:02",
                         "type": "interface",
@@ -321,7 +321,7 @@
                         "id": "00:00:00:00:00:00:00:02:3",
                         "link": "4d42dc0852278accac7d9df15418f6d921db160b13d674029a87cef1b5f67f30",
                         "lldp": true,
-                        "mac": "52:58:f1:d3:1d:f2",
+                        "mac": "ae:2a:14:42:4f:ce",
                         "metadata": {},
                         "name": "s2-eth3",
                         "nni": true,
@@ -329,17 +329,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 2112,
+                            "rx_bytes": 180,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 15,
-                            "tx_bytes": 2112,
+                            "rx_packets": 2,
+                            "tx_bytes": 90,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 1
                         },
                         "switch": "00:00:00:00:00:00:00:02",
                         "type": "interface",
@@ -351,7 +351,7 @@
                         "id": "00:00:00:00:00:00:00:02:4294967294",
                         "link": "",
                         "lldp": true,
-                        "mac": "e2:72:fd:ab:04:4d",
+                        "mac": "8e:38:41:21:6b:46",
                         "metadata": {},
                         "name": "s2",
                         "nni": false,
@@ -389,7 +389,7 @@
             },
             "00:00:00:00:00:00:00:03": {
                 "active": true,
-                "connection": "127.0.0.1:39188",
+                "connection": "127.0.0.1:39196",
                 "data_path": "s3",
                 "dpid": "00:00:00:00:00:00:00:03",
                 "enabled": true,
@@ -402,7 +402,7 @@
                         "id": "00:00:00:00:00:00:00:03:1",
                         "link": "",
                         "lldp": true,
-                        "mac": "6a:a7:11:7b:ab:dd",
+                        "mac": "36:0b:4b:74:82:f6",
                         "metadata": {},
                         "name": "s3-eth1",
                         "nni": false,
@@ -410,17 +410,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 516,
+                            "rx_bytes": 176,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 6,
-                            "tx_bytes": 2112,
+                            "rx_packets": 2,
+                            "tx_bytes": 180,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 2
                         },
                         "switch": "00:00:00:00:00:00:00:03",
                         "type": "interface",
@@ -432,7 +432,7 @@
                         "id": "00:00:00:00:00:00:00:03:2",
                         "link": "4d42dc0852278accac7d9df15418f6d921db160b13d674029a87cef1b5f67f30",
                         "lldp": true,
-                        "mac": "8e:5b:a9:86:21:7e",
+                        "mac": "b2:28:bc:9c:63:71",
                         "metadata": {},
                         "name": "s3-eth2",
                         "nni": true,
@@ -440,17 +440,17 @@
                         "speed": 1250000000.0,
                         "stats": {
                             "collisions": 0,
-                            "rx_bytes": 2112,
+                            "rx_bytes": 90,
                             "rx_crc_err": 0,
                             "rx_dropped": 0,
                             "rx_errors": 0,
                             "rx_frame_err": 0,
                             "rx_over_err": 0,
-                            "rx_packets": 15,
-                            "tx_bytes": 2112,
+                            "rx_packets": 1,
+                            "tx_bytes": 180,
                             "tx_dropped": 0,
                             "tx_errors": 0,
-                            "tx_packets": 15
+                            "tx_packets": 2
                         },
                         "switch": "00:00:00:00:00:00:00:03",
                         "type": "interface",
@@ -462,7 +462,7 @@
                         "id": "00:00:00:00:00:00:00:03:4294967294",
                         "link": "",
                         "lldp": true,
-                        "mac": "8e:89:4c:e7:0c:46",
+                        "mac": "1e:bb:57:78:46:45",
                         "metadata": {},
                         "name": "s3",
                         "nni": false,

```

current `topology` HEAD commit 9c9752ce2da7fc30b95c91d9fcf146b982430952 that I'm running locally, I have't bisected, I'll investigate.

**Comment by italovalcy** [comment]
@viniarck I think this is related to the way of_core handles OFPT_PORT_STATUS (https://github.com/kytos-ng/of_core/blob/master/main.py#L644-L650 and https://github.com/kytos-ng/of_core/blob/master/main.py#L666-L672), which means a new Interface is created, so the reference at the Link object gets wrong and as a consequence the active status gets unsync (the diff you showed above). Additionally, the PORT_STATUS removed should not remove the interface from the switch, but just deactivate it instead.

Recently, @ajoaoff  has created a method `Switch.update_or_create_interface()` that should be used. We did that for PORT_STATS (PR kytos-ng/of_core#21) but not for PORT_STATUS (my bad).

**Comment by viniarck** [comment]
Closing this issue since it has been fixed on PR #38
